### PR TITLE
Fix extraction.py's create_update

### PR DESCRIPTION
**Pull request by dotfloat**
When refactoring for FastAPI, the create_update function was incorrectly
transferred. The lack of ensemble updates came from a bug where a
function was inlined naively, causing an early return.

The usage of the term 'misfit' is incorrect but I've chosen not to touch
it. It should rather be called an update or a synonym thereof.

### CODE REVIEW

File: ert_shared/libres_facade.py [hunk 1]
```diff
@@ -11,21 +11,21 @@ class LibresFacade(object):
     """Facade for libres inside ERT."""
 
     def __init__(self, enkf_main):
-        self._enkf_main = enkf_main
+        self.enkf_main = enkf_main
```
Comment (oysteoh) [inline ert_shared/libres_facade.py:14]: why the change from indicating private to public?  the whole purpose of the facade is to wrap enkf_main and we don't want anyone to use it directly 🤔

**Review by oysteoh — CHANGES_REQUESTED** [review]
you need to black your files, and - i have a question wrt the namechange of `_enkf_main` to `enkf_main` in the facade. And - if we were to decide this is needed / sane - then i would suggest doing this in a separate commit first to avoid the actual change in logic being disturbed by all the lines regarding name-change.

File: ert_shared/libres_facade.py (hunk 1 above)
Comment (dotfloat) [reply ert_shared/libres_facade.py:14]: That is perhaps correct. The reason is that extraction uses `ERT.ert` and I needed to replace that with something else, but it is probably better to dig deeper and get rid of that line.

Comment (oysteoh) [reply ert_shared/libres_facade.py:14]: yes - the ERT.ert thing.... 🙈  But at least it's hidden inside the facade... But as stated in the message below - if you feel the need for this to change - we can discuss ! And if we decide this is the way to do it - then at least i want two commits - one which contains the renaming and one with the actual logic 👍

**Review by oysteoh — APPROVED** [review]
🚀
